tests/io/test_tags.py

```diff
@@ -82,292 +123,455 @@ def test_tags_create_category_success(api, tagcat):
 def test_tags_edit_category_value_typeerror(api):
+    '''
+    test to raise exception when type of value param does not match the expected type.
+    '''
     with pytest.raises(TypeError):
         api.tags.edit_category(uuid.uuid4(), value=1)
 
 @pytest.mark.vcr()
 def test_tags_edit_category_description_typeerror(api):
+    '''
+    test to raise exception when type of description param does not match the expected type.
+    '''
     with pytest.raises(TypeError):
         api.tags.edit_category(uuid.uuid4(), description=1)
 
 @pytest.mark.vcr()
 def test_tags_edit_category_success(api, tagcat):
-    t = api.tags.edit_category(tagcat['uuid'], name='Edited')
-    assert isinstance(t, dict)
-    check(t, 'uuid', 'uuid')
-    check(t, 'created_at', 'datetime')
-    check(t, 'created_by', str)
-    check(t, 'updated_at', 'datetime')
-    check(t, 'updated_by', str)
-    check(t, 'name', str)
+    '''
+    test to edit tag category information.
+    '''
+    resp = api.tags.edit_category(tagcat['uuid'], name='Edited')
+    assert isinstance(resp, dict)
+    check(resp, 'uuid', 'uuid')
+    check(resp, 'created_at', 'datetime')
+    check(resp, 'created_by', str)
+    check(resp, 'updated_at', 'datetime')
+    check(resp, 'updated_by', str)
+    check(resp, 'name', str)
     #check(t, 'description', str, allow_none=True)
-    check(t, 'reserved', bool)
-    assert t['name'] == 'Edited'
+    check(resp, 'reserved', bool)
+    assert resp['name'] == 'Edited'
 
 def test_tags_list_constructor_filter_type_typeerror(api):
+    '''
+    test to raise exception when type of filter_type param does not match the expected type.
+    '''
     with pytest.raises(TypeError):
         api.tags._tag_list_constructor([],
             api.tags._filterset_tags, True, None)
 
 def test_tags_list_constructor_filter_type_unexpectedvalueerror(api):
+    '''
+    test to raise exception when filter_type param value does not match the choices.
+    '''
     with pytest.raises(UnexpectedValueError):
         api.tags._tag_list_constructor([],
             api.tags._filterset_tags, 'nadda', None)
 
 def test_tags_list_constructor_filter_type_success(api):
+    '''
+    test to check filter_type param in tags_list_constructor method.
+    '''
     resp = api.tags._tag_list_constructor([],
         api.tags._filterset_tags, 'and', None)
     assert resp['ft'] == 'AND'
 
 def test_tags_list_constructor_sort_typeerror(api):
+    '''
+    test to raise exception when type of sort param does not match the expected type.
+    '''
     with pytest.raises(TypeError):
         api.tags._tag_list_constructor([],
             api.tags._filterset_tags, None, 1)
 
 def test_tags_list_constructor_sort_unexpectedvalueerror(api):
+    '''
+    test to raise exception when sort param value does not match the choices.
+    '''
     with pytest.raises(UnexpectedValueError):
         api.tags._tag_list_constructor([],
             api.tags._filterset_tags, None, (('something_else'),))
```

Comment:
Access to a protected member of the _tags_list_constructor

Reply:
Addressed above comment in next commit

Reply:
Addressed above comment in next commit